lib/filerotatetransport.js

```diff
@@ -0,0 +1,128 @@
+var winston = require('winston'),
+    util = require('util'),
+    path = require('path'),
+    fs = require('fs'),
+    async = require('async'),
+    glob = require('glob'),
+    debug = require('debug')('hmpo:filerotatelogger');
+
+var FileRotate = function (options) {
+    options = options || {};
+
+    if (options.maxsize && options.dateRotate) {
+        throw new Error('Cannot set maxsize and dateRotate together');
+    }
+
+    winston.transports.File.apply(this, arguments);
+
+    this.dateRotate = options.dateRotate;
+    if (this.dateRotate) {
+        this._dateRotateUpdateDay();
+    }
+};
+
+util.inherits(FileRotate, winston.transports.File);
+
+FileRotate.prototype.name = 'filerotate';
+
+FileRotate.prototype._dateRotateUpdateDay = function () {
+    var d = new Date();
+    var year = d.getUTCFullYear();
+    var month = d.getUTCMonth();
+    var day = d.getUTCDate();
+    this._dateRotateStartTime = Date.UTC(year, month, day, 0, 0, 0, 0);
+    this._dateRotateEndTime = Date.UTC(year, month, day + 1, 0, 0, 0, 0);
+    debug('new end date', this._dateRotateEndTime);
+};
+
+FileRotate.prototype.open = function (callback) {
+    var self  = this;
+    if (!this.opening && this.dateRotate && Date.now() >= this._dateRotateEndTime) {
+        debug('Needs new log file');
+
+        // Don't allow another open() while rotating files.
+        // This also will get set in _createStream
+        this.opening = true;
+
+        this._dateRotateLog(function () {
+            self._dateRotateRemoveOldFiles(function () {
```

Comment:
This was in the style of the parent file transport, but yes  i can update to be in a binding style instead